### PR TITLE
Add a puppeteer testkit bundle

### PR DESCRIPTION
**Pull request by ohana54**
### 🔦 Summary
TL;DR - This will allow Sled test writers to use the WSR puppeteer drivers in their tests.

When writing Sled tests, you can only use packages that exist in the lambda environment.
We want to create a bundle for the puppeteer drivers which we will upload to the lambda environment so it will be available during test execution.
We could do nothing, and upload the entire WSR module, but it's too big (around 400MB) and we are limited in space.

cc @tompere

### CODE REVIEW

File: testkit/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        use: {
+          loader: 'babel-loader',
+        },
+      },
+    ],
+  },
+  plugins: [
+    // Puppeteer drivers don't use styles, so we want to ignore them
+    // by replacing them with empty objects
+    new webpack.NormalModuleReplacementPlugin(
```
Comment (shlomitc) [inline testkit/webpack.config.js]: why is this needed? are there any drivers that currently do import scss files?

File: package.json [hunk 2]
```diff
@@ -39,7 +39,8 @@
     "test:types": "tsc --noEmit -p tsconfig.declarations.json",
     "copy-public-styles": "cp src/*.scss .",
     "prebuild": "npm run generate-testkit-exports",
-    "build": "npm run lint && npm run proxy-wix-ui-icons && npm run transpile && npm run storybook:build",
+    "build": "npm run transpile && npm run storybook:build && npm run build-puppeteer-testkit",
```
Comment (shlomitc) [inline package.json]: maybe it can run in parallel to other tasks instead of sequential?

File: testkit/webpack.config.js (hunk 1 above)
Comment (ohana54) [reply testkit/webpack.config.js]: Unfortunately yes - but not in the way you think (as you showed me, no driver is actually directly importing styles).
The problem is something like this:
1. puppeteer.js (the entry point) imports a **uni** button driver
2. The uni driver imports scss files somewhere in its dependency graph
So if I have this kind of import:
`import { buttonNextDriverFactory } from 'wix-ui-core/drivers/unidriver';`
Webpack will try to import **everything** exported by `wix-ui-core/drivers/unidriver`

Assuming we don't need styles, like we discussed, I thought the safest easiest thing is to just ignore these imports. I'm open to more suggestions :)

File: package.json (hunk 2 above)
Comment (ohana54) [reply package.json]: Done

**Comment by ohana54** [comment]
@yavorsky please let us know when we can have your input, everything else is done :)
Thanks!

File: package.json [hunk 3]
```diff
@@ -39,7 +39,8 @@
     "test:types": "tsc --noEmit -p tsconfig.declarations.json",
     "copy-public-styles": "cp src/*.scss .",
     "prebuild": "npm run generate-testkit-exports",
-    "build": "npm run lint && npm run proxy-wix-ui-icons && npm run transpile && npm run storybook:build",
```
Comment (yavorsky) [inline package.json]: @shlomitc are we ok with removing `lint` from build script?

File: testkit/webpack.config.js (hunk 1 above)
Comment (yavorsky) [reply testkit/webpack.config.js]:
```
So if I have this kind of import:
import { buttonNextDriverFactory } from 'wix-ui-core/drivers/unidriver';
Webpack will try to import everything exported by wix-ui-core/drivers/unidriver
```

It shouldn't with `sideEffects: false` in module's package.json. And if this module won't be transpiled with babel to commonjs modules.

File: testkit/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      /\.(s?css)$/,
+      path.join(__dirname, './fake-style-module.js'),
+    ),
+  ],
+  mode: 'development',
```
Comment (yavorsky) [inline testkit/webpack.config.js]: should it be `development` for e2e testing ?

File: testkit/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    path: path.resolve('./dist/testkit'),
+    filename: 'puppeteer-testkit-bundle.js',
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'babel-loader',
```
Comment (yavorsky) [inline testkit/webpack.config.js]: maybe we can use options here and pass `babel-preset-env` with `last 2 chrome versions` so it will be compiled faster just for latest chrome?

File: package.json (hunk 3 above)
Comment (ohana54) [reply package.json]: Looks like a bad merge? I never intended to remove anything, I'll fix

File: testkit/webpack.config.js (hunk 5 above)
Comment (ohana54) [reply testkit/webpack.config.js]: I'm using the same babel config as the rest of the app, I don't know if I can diverge. @shlomitc ?

File: testkit/webpack.config.js (hunk 1 above)
Comment (ohana54) [reply testkit/webpack.config.js]: But it does :)

File: testkit/webpack.config.js (hunk 4 above)
Comment (ohana54) [reply testkit/webpack.config.js]: It's the driver, so I personally prefer it won't be minified in case I need to step into.

File: testkit/webpack.config.js (hunk 5 above)
Comment (shlomitc) [reply testkit/webpack.config.js]: sounds like an optimization, not really sure it is a must, so depends on you guys to decide

File: testkit/webpack.config.js (hunk 4 above)
Comment (shlomitc) [reply testkit/webpack.config.js]: makes sense

File: testkit/webpack.config.js (hunk 1 above)
Comment (shlomitc) [reply testkit/webpack.config.js]: @yavorsky - maybe the right thing will not be taking the drivers from the cherry-picked `wix-ui-core/drivers` entry? just take them from `dist/src`?

Comment (yavorsky) [reply testkit/webpack.config.js]: Yeah. It's not working b/c unidriver.js has commonjs modules.
We can:
- move it to src, so we can use both `dist/src/drivers` and `dist/es/src/drivers`.
-  update the script, to create `es` files in own directory: `import { buttonNextDriverFactory } from 'wix-ui-core/drivers/es/unidriver'`

File: testkit/webpack.config.js (hunk 5 above)
Comment (ohana54) [reply testkit/webpack.config.js]: I prefer not to add the complication of a separate build configuration, the build output is exactly the same as other drivers we supply to consumers.

**Comment by ohana54** [comment]
@yavorsky and @shlomitc thank you for the review and the insightful comments!
